pkg/apis/sources/v1alpha1/sinkbinding_lifecycle.go

```diff
@@ -49,3 +72,49 @@ func (fbs *SinkBindingStatus) MarkBindingUnavailable(reason, message string) {
 func (fbs *SinkBindingStatus) MarkBindingAvailable() {
 	sbCondSet.Manage(fbs).MarkTrue(SinkBindingConditionReady)
 }
+
+// Do implements psbinding.Bindable
+func (fb *SinkBinding) Do(ctx context.Context, ps *duckv1.WithPod) {
+	// First undo so that we can just unconditionally append below.
+	fb.Undo(ctx, ps)
+
+	uri := GetSinkURI(ctx)
+	if uri == nil {
+		logging.FromContext(ctx).Error(fmt.Sprintf("No sink URI associated with context for %+v", fb))
+		return
+	}
+
+	spec := ps.Spec.Template.Spec
+	for i := range spec.InitContainers {
+		spec.InitContainers[i].Env = append(spec.InitContainers[i].Env, corev1.EnvVar{
+			Name:  "K_SINK",
```

Comment:
How would you feel about this being set-able from the spec of the binding? And default to `K_SINK` if not set. I think there will be a lot of existing code that uses `SINK`, `TARGET`, `OUTPUT`, etc that we could let folks just hook into (like code that was written before knative)

Reply:
We can do that as a followup...